### PR TITLE
Change add task implemetation

### PR DESCRIPTION
**Pull request by ValeriiSudakov**
Moved add task and add subtask implementation from TaskService to TaskRepository

### CODE REVIEW

File: src/API/TaskService.cpp [hunk 1]
```diff
@@ -8,21 +8,13 @@
...
 
 AddTaskResult TaskService::AddTask(const TaskDTO& task, const Task::Priority& priority){
-  auto newTask = tasksRepository_.GetTaskStorage().AddTask(task.GetTask(), priority);
-  if (newTask.has_value()) {
-    tasksRepository_.GetTaskView().AddTask(newTask.value());
-    return AddTaskResult(true);
-  }
-  return AddTaskResult(AddTaskResult::ErrorType::NOT_ENOUGH_FREE_MEMORY, false);
+  AddTaskResult addNewTask = tasksRepository_.AddTask(task, priority);
+  return addNewTask.success_;
```
Comment (denys-hrulov) [inline src/API/TaskService.cpp:12]: I thins these two lines can be replaced by
return tasksRepository_.AddTask(task, priority).success_

By then it's obvious that Service is actually doing nothing.

**Review by denys-hrulov — APPROVED** [review]

File: src/API/TaskService.cpp (hunk 1 above)
Comment (ValeriiSudakov) [reply src/API/TaskService.cpp:12]: It ask taskRepository to add new task to storage. All TaskService does is ask another classes to do smth.
And yeah, I can merge these 2 lines in 1

Comment (ValeriiSudakov) [reply src/API/TaskService.cpp:12]: It asks taskRepository to add new task to storage. All TaskService does is ask another classes to do smth.
And yeah, I can merge these 2 lines in 1

Comment (ValeriiSudakov) [reply src/API/TaskService.cpp:12]: Also 11 line shows return type of AddTask. I think this will impruve code readabilty

Comment (vladimir-moiseiev) [inline src/API/TaskService.cpp:12]: Why are we returning private member directly?

File: src/Memory_Model/Storage/TaskView.cpp [hunk 2]
```diff
@@ -3,7 +3,7 @@
 //
 
 #include "TaskView.h"
-#include "Memory_Model/Date.h"
+#include "Memory_Model/Date/Date.h"
```
Comment (vladimir-moiseiev) [inline src/Memory_Model/Storage/TaskView.cpp:6]: Why is rename is the part of this review?

**Pull request merged**